client-data/tools/pencil/pencil.js

```diff
@@ -153,11 +171,20 @@
 			"name": "White-out",
 			"icon": "tools/pencil/whiteout_tape.svg",
 			"active": false,
-			"switch": stopLine,
+			"switch": function() {
+				stopLine();
+				toggleSize();
+			},
+		},
+		"onquit": function() {
+			//When switching to another tool, restore Pencil and its size
+			if (pencilTool.secondary.active) {
+				Tools.change("Pencil");
+			}
 		},
```

Comment:
Yes, I don't think we want to switch back to pencil when changing tools. People can do it manually if they want; but I think automatically switching back to primary only for the pencil tool is confusing.

```suggestion
```

Reply:
Agreed, that was confusing. But here's what I wanted to achieve by that. Imagine I'm using a Pencil, then switch to White-out, set different size, correct my drawing, then switch to Rectangle. Now my expectation Rectangle would have the same line width Pencil had, but instead it inherits White-out size. In the latest commit I'm addressing that by treating Pencil line width and all other tool line widths as "Drawing" line width, and White-out has its individual line width.

Reply:
I don't think it's surprising that WBO **doesn't change** the line width when you switch between tools.
We may want to have a separate width for each tool, but this is a different question.